config/routes.rb

```diff
@@ -1377,6 +1377,26 @@
       )
     },
 
+    :physical_storage    =>  {
+      :get  =>  %w(
+        download_data
+        download_summary_pdf
+        show_list
+        show
+      ) + compare_get,
+
+      :post   =>  %w(
+        button
+        listnav_search_selected
+        show_list
+        create
```

Comment:
What about all these create and update routes? do you need these?

Also, if the controller + action name match the feature name, then this would work: https://github.com/ManageIQ/manageiq-ui-classic/blob/master/app/controllers/application_controller.rb#L1172

The idea is that people forget to add the `assert_rbac(...)` to their action so there's a second checking mechanism in place for all the actions.

Of course you might need actions that don't check RBAC (but those should not be POST for sure) and sure you might want to pass throught the for all mechanism and just use your `assert_rbac` call in your action.

`physical_storage_' + 'create' looks good. It might be a feature name that you might have. But '`physical_storage_' + 'create_del' looks weird.

Reply:
> What about all these create and update routes? do you need these?

@martinpovolny Currently no, I thought that some of these could be used in the future, but I'm going to remove them and leave only the necessary for now. Also, and correct me if I'm wrong, I just noticed that some routes in `compare_get` might not be supported in this case, so to be safe I'll remove it too. If they'll be needed in the future, we can simply add it here.

Reply:
Unless you are adding the compare capability for the new entity you don't need the `compare_get` set of routes.
Please, do not add routes that you do not need ;-)